infra/bicep/core/ai/ai-hub-secure.bicep

```diff
@@ -42,17 +42,36 @@ param addRoleAssignments bool = true
 param userObjectId string = ''
 param userObjectType string = 'User'
 
-@description('The object ID of the application identity to be granted necessary role assignments to access the Azure AI Hub.')
-param managedIdentityId string = ''
+// @description('The object ID of the application identity to be granted necessary role assignments to access the Azure AI Hub.')
+// param managedIdentityResourceId string = ''
+@description('The principal ID of the application identity to be granted necessary role assignments to access the Azure AI Hub.')
+param managedIdentityPrincipalId string = ''
 param managedIdentityType string = 'ServicePrincipal'
 
-resource aiHub 'Microsoft.MachineLearningServices/workspaces@2024-04-01-preview' = {
+@description('Name AI Search resource')
+param aiSearchName string
+param aiSearchResourceGroupName string = resourceGroup().name
+
+var aiServiceConnectionName = '${aiHubName}-connection-AIService'
+var searchConnectionName  = '${aiHubName}-connection-AISearch'
+
+resource aiSearch 'Microsoft.Search/searchServices@2020-03-13' existing = {
+  name: aiSearchName
+  scope: resourceGroup(aiSearchResourceGroupName)
+}
+
+resource aiHub 'Microsoft.MachineLearningServices/workspaces@2024-10-01' = {
   name: aiHubName
   location: location
   tags: tags
   kind: 'Hub'
   identity: {
     type: 'SystemAssigned'
+    // tried this but it keeps getting a 400 error with not other details...?
```

Comment:
user assigned works for me

Reply:
Thought I had a problem... never got back to trying that again